dartagnan/src/main/java/com/dat3m/dartagnan/expression/utils/IntegerHelper.java

```diff
@@ -0,0 +1,225 @@
+package com.dat3m.dartagnan.expression.utils;
+
+import com.google.common.base.Preconditions;
+
+import java.math.BigInteger;
+import java.util.function.BiFunction;
+
+public class IntegerHelper {
+
+    public static final boolean SIGNED_NORMALIZATION = true;
```

Comment:
Shouldn't this be moved to `GlobalSetting` to keep all such configurable options centralized?

Reply:
Possibly. But honestly, I'm not too sure if we even need this option at all. I kept most code to work over non-normalized values (i.e., it is correct either way).
It is hard to tell what is best to do.

Reply:
If there is no clear win with keeping this, I would simply get rid of it.

Reply:
I removed it.